### PR TITLE
Dont use raw ptr for nodehandle

### PR DESCRIPTION
**Pull request by Dale-Koenig**

### CODE REVIEW

File: supplementary/alica_ros1/alica_ros_proxy/src/communication/AlicaRosCommunication.cpp [hunk 1]
```diff
@@ -41,36 +41,36 @@ const std::string presenceAnnouncementTopic = "/AlicaEngine/AgentAnnouncement";
 AlicaRosCommunication::AlicaRosCommunication(const AlicaCommunicationHandlers& callbacks, ros::CallbackQueue& cb_queue)
         : IAlicaCommunication(callbacks)
         , _callbackQueue(cb_queue)
+        , _nh()
```
Comment (ntfshard) [inline supplementary/alica_ros1/alica_ros_proxy/src/communication/AlicaRosCommunication.cpp:44]: nitpick: redundant

**Review by ntfshard — APPROVED** [review]

File: supplementary/alica_ros1/alica_ros_proxy/src/communication/AlicaRosCommunication.cpp [hunk 2]
```diff
@@ -80,8 +80,7 @@ AlicaRosCommunication::~AlicaRosCommunication()
     _planTreeInfoSubscriber.shutdown();
     _syncReadySubscriber.shutdown();
     _syncTalkSubscriber.shutdown();
-    _rosNode->shutdown();
-    delete _rosNode;
+    _nh.shutdown();
```
Comment (rogeriofonteles) [inline supplementary/alica_ros1/alica_ros_proxy/src/communication/AlicaRosCommunication.cpp:83]: since this is not a pointer anymore, is this necessary?

Comment (ntfshard) [reply supplementary/alica_ros1/alica_ros_proxy/src/communication/AlicaRosCommunication.cpp:83]: `delete` also calls `destructor`, in other words it's not related to this question
but calling `shutdown` prevents callback invocation in destroyed object

Comment (Dale-Koenig) [reply supplementary/alica_ros1/alica_ros_proxy/src/communication/AlicaRosCommunication.cpp:83]: I guess we need either this, the subscriber shutdowns, or to be careful with declaration order (and I'd rather not rely on the last) but we might not need both this and the subscriber explicit shutdowns

**Review by triccyx — APPROVED** [review]

**Pull request merged**